csharp-linq/LINQInnerJoin/LinqInnerJoin/EnrolmentHelper.cs

```diff
@@ -0,0 +1,176 @@
+        {
+            var categories = Category.GetDummyCourseCategories();
+
+            var courseWithCategoryName = courses.Join(
+                categories,
+                course => course.CategoryId,
+                category => category.Id,
+                (course, category) => new { course.Id, course.Name, CategoryName = category.Name }
+            );
+
+            return courseWithCategoryName;
+        }
+
+        public static void PrintCoursesWithCategory(IEnumerable<dynamic> courses)
+        {
+            foreach (var course in courses)
+            {
+                Console.WriteLine(
+                    $"Id: {course.Id} Course: {course.Name}  Category: {course.CategoryName}"
+                );
+            }
+        }
+
+        public static void PrintDummyCourses()
+        {
+            foreach (var course in Course.GetDummyCourses())
+            {
+                Console.WriteLine(
+                    $"Id: {course.Id} Course: {course.Name}  CategoryId: {course.CategoryId} "
+                );
+            }
+        }
+
+        public static void PrintDummyCategories()
+        {
+            foreach (var category in Category.GetDummyCourseCategories())
+            {
+                Console.WriteLine($"Id: {category.Id} Category: {category.Name}");
+            }
+        }
+
+        public static void PrintEnrolments(IEnumerable<dynamic> enrolments)
+        {
+            foreach (var enrolment in enrolments)
+            {
+                Console.WriteLine(
+                    $"Id: {enrolment.Id}, Student Name: {enrolment.StudentName}, Course: {enrolment.CourseName} \n"
+                );
+            }
+        }
+
+        public static void PrintEnrolmentsGroup(IEnumerable<dynamic> enrolmentGroups)
+        {
+            foreach (var group in enrolmentGroups)
+            {
+                Console.WriteLine($"Enrolments for Course Id: {group.Key} \n");
+                foreach (var enrolment in group)
+                {
+                    Console.WriteLine(
+                        $"Id: {enrolment.Id}, Student Name: {enrolment.StudentName}, Course: {enrolment.CourseName} \n"
+                    );
+                }
+            }
+        }
+
+        private static IEnumerable<dynamic> GetEnrolmentsRelation(
+            IEnumerable<Enrolment> enrolments,
+            IEnumerable<Student> students,
+            IEnumerable<Course> courses
+        )
+        {
+            return enrolments
+                .Join(
+                    students,
+                    enrolmentLevel1 => enrolmentLevel1.StudentId,
+                    student => student.Id,
+                    (enrolmentLevel1, student) => new { enrolmentLevel1, student }
+                )
+                .Join(
+                    courses,
+                    enrolmentLevel2 => enrolmentLevel2.enrolmentLevel1.CourseId,
```

Comment:
here instead of `enrolmentLevel2` it can be changed to `enrolmentStudentRelation` 
And then alongside with earlier change we will have `enrolmentStudentRelation => enrolmentStudentRelation.enrolment.CourseId`